syllabus/core/stat_recorder.py

```diff
@@ -34,45 +31,96 @@ def record(self, episode_return: float, episode_length: int, episode_task, env_i
         """
         Records the length and return of an episode for a given task.
 
-        :param task: Identifier for the task
+        :param episode_task: Identifier for the task
         :param episode_length: Length of the episode, i.e. the total number of steps taken during the episode
         :param episodic_return: Total return for the episode
         """
 
         if episode_task in self.tasks:
-            self.records[episode_task].append({
-                "r": episode_return,
-                "l": episode_length,
-                "env_id": env_id
-            })
-            self.stats[episode_task]['mean_r'] = np.mean([record["r"] for record in self.records[episode_task]])
-            self.stats[episode_task]['var_r'] = np.var([record["r"] for record in self.records[episode_task]])
-            self.stats[episode_task]['mean_l'] = np.mean([record["l"] for record in self.records[episode_task]])
-            self.stats[episode_task]['var_l'] = np.var([record["l"] for record in self.records[episode_task]])
+            if self.calc_past_N:
+                self.records[episode_task].append({
+                    "r": episode_return,
+                    "l": episode_length,
+                    "env_id": env_id
+                })
+                self.records[episode_task] = self.records[episode_task][-self.calc_past_N:]
```

Comment:
It would be more efficient to implement this with a separate queue for return, length, and id's https://docs.python.org/3/library/collections.html#collections.deque